Modules/include/VersionInfoProvider.h

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#pragma once
+
+#include "ApplicationModule.h"
+#include "ConfigReader.h"
+#include "ScalarAccessor.h"
+#include "VersionInfo.h"
+#include "boost/format.hpp"
+
+namespace ChimeraTK::VersionInfo {
+
+  extern const int major;
+  extern const int minor;
+  extern const int applicationPatch;
+
+} // namespace ChimeraTK::VersionInfo
```

Comment:
I just realised that we screwed up the generated VersionInfo.h file, sorry for missing this earlier.

When defining exported global variables in C/C++, they need to be declared in a header file with the `extern` keyword and defined in a compiled .c file (and also the value needs to go there). Otherwise the ODR (one definition rule) is violated, if the header file is included multiple times.

As an alternative, because we are using constants only, we can make the variables `constexpr`, in which case it is ok to define them multiple times, but in that case we do not need the `extern` declaration at all, just the definition in the header file is sufficient.

So I propose to remove these lines here entirely and touch the project-template again and change `const` into `constexpr` there. As long the project-template is not yet propagated to any projects, we can even force-push this to the master branch to avoid a cluttered history (of all our projects!).

Reply:
> we can even force-push

Branch is protected, changes must be made through a pull request.

Reply:
That is not a final restriction ;-)